libraries/AP_Math/matrix3.cpp

```diff
@@ -228,6 +228,42 @@ T Matrix3<T>::det() const
 }
 
 template <typename T>
+Matrix3<T> Matrix3<T>::inverse(bool& success) const
```

Comment:
It makes sense. Actually, that was one of my alternatives, but I decided to try to be consistent with the way transpose methods work. I'll change it to the way you suggested.

> Then just make sure it still works with *this == other.

Not sure what you mean here.